src/routes/Products.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+import { useFetchProducts } from "../store/products/products.hooks";
+import { useEffect } from "react";
+import { Link } from "react-router-dom";
+
+const Products: React.FC = () => {
+  const { submit: fetchProducts, data = [], error } = useFetchProducts();
+
+  useEffect(() => {
+    fetchProducts();
```

Comment:
A hook, `useEffect` is used to make an API request `fetchProduct` at the top level within this script?

Reply:
Exactly right. useEffect takes a function and an array (dependency array) and basically says "run this function any time values in the dependency array change"

https://reactjs.org/docs/hooks-effect.html

So in order to run a function once, `useEffect(() => doSomething(), [])` you pass it an empty dependency array.

If you don't use `useEffect` , and just called `fetchProduct` at the top level within the script, then `fetchProduct` would be called every single time the component renders -- which would NOT be desirable behaviour.

Whilst you're learning react, an interesting thing to do is put a `console.log("Component rendering")` in the component function, and open to browser developer console to see how many times a component is rendering.